Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift

```diff
@@ -13,42 +13,44 @@ extension LogtoClient {
     func signInWithBrowser<AuthSession: LogtoAuthSession>(
         authSessionType _: AuthSession.Type,
         redirectUri: String
-    ) async throws -> Errors.SignIn? {
+    ) async throws {
         guard let redirectUri = URL(string: redirectUri) else {
-            return (Errors.SignIn(type: .unableToConstructRedirectUri, innerError: nil))
+            throw (Errors.SignIn(type: .unableToConstructRedirectUri, innerError: nil))
         }
 
         let oidcConfig = try await fetchOidcConfig()
 
-        return await withCheckedContinuation { [self] continuation in
-            let session = AuthSession(
-                logtoConfig: logtoConfig,
-                oidcConfig: oidcConfig,
-                redirectUri: redirectUri,
-                socialPlugins: socialPlugins
-            ) { [self] in
-                switch $0 {
-                case let .failure(error):
-                    continuation.resume(returning: error)
-                case let .success(response):
-                    idToken = response.idToken
-                    refreshToken = response.refreshToken
-                    accessTokenMap[buildAccessTokenKey(for: nil, scopes: [])] = AccessToken(
-                        token: response.accessToken,
-                        scope: response.scope,
-                        expiresAt: Date().timeIntervalSince1970 + TimeInterval(response.expiresIn)
-                    )
-                    continuation.resume(returning: nil)
-                }
-            }
-
-            session.start()
+        let session = AuthSession(
+            logtoConfig: logtoConfig,
+            oidcConfig: oidcConfig,
+            redirectUri: redirectUri,
+            socialPlugins: socialPlugins
+        )
+
+        let response = try await session.start()
+
+        idToken = response.idToken
+        refreshToken = response.refreshToken
+        accessTokenMap[buildAccessTokenKey(for: nil, scopes: [])] = AccessToken(
+            token: response.accessToken,
+            scope: response.scope,
+            expiresAt: Date().timeIntervalSince1970 + TimeInterval(response.expiresIn)
+        )
+
+        if let idToken = idToken {
+            let jwks = try await fetchJwkSet()
+            try LogtoUtilities.verifyIdToken(
+                idToken,
+                issuer: oidcConfig.issuer,
+                clientId: logtoConfig.clientId,
+                jwks: jwks
+            )
```

Comment:
We should verify the Id token first before we save the access token.

Reply:
make sense. updated, ptal